shm/src/Ship.cpp

```diff
@@ -37,8 +37,13 @@ void Ship::setName(const std::string& name) {
 }
 
 std::shared_ptr<Cargo> Ship::getCargo(size_t index) const {
-    if (cargo_.size() <= index) {
+    if (index>= cargos_.size()) {
+        std::cerr << "Invalid cargo\n";
```

Comment:
output not here

Reply:
this is anyway the function we need to rewrite if you see the review, but i don't know how